ambry-router/src/main/java/com.github.ambry.router/CryptoJobHandler.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.utils.Utils;
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Responsible for handling crypto jobs. {@link Router} instantiates this {@link CryptoJobHandler} and gives a
+ * reference of the same to PutManager, GetBlobOperation and GetBlobInfoOperation which will add jobs via
+ * {@link #submitJob(CryptoJob)}. On close, all pending jobs will be processed(either successfully or exception is set)
+ * and any new jobs submitted after close will be ignored.
+ */
+class CryptoJobHandler implements Closeable {
```

Comment:
Any reason we're doing an ad-hoc thread pool implementation instead of using ExecutorService?

Reply:
@nsivabalan